src/main/cpp/src/regex_rewrite_utils.cu

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/detail/utf8.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/transform.h>
+
+namespace spark_rapids_jni {
+
+namespace {
+
+struct literal_range_pattern_fn {
+  __device__ bool operator()(
+    cudf::string_view d_string, cudf::string_view d_prefix, int range_len, int start, int end)
+  {
+    int const n = d_string.length(), m = d_prefix.length();
+    for (int i = 0; i <= n - m - range_len; i++) {
+      bool match = true;
+      for (int j = 0; j < m; j++) {
+        if (d_string[i + j] != d_prefix[j]) {
+          match = false;
+          break;
+        }
+      }
+      if (match) {
+        for (int j = 0; j < range_len; j++) {
+          auto code_point = cudf::strings::detail::utf8_to_codepoint(d_string[i + m + j]);
+          if (code_point < start || code_point > end) {
+            match = false;
+            break;
+          }
+        }
+        if (match) { return true; }
+      }
+    }
+    return false;
+  }
+};
+
+std::unique_ptr<cudf::column> find_literal_range_pattern(cudf::strings_column_view const& strings,
+                                                         cudf::string_scalar const& prefix,
+                                                         int const range_len,
+                                                         int const start,
+                                                         int const end,
+                                                         rmm::cuda_stream_view stream,
+                                                         rmm::device_async_resource_ref mr)
+{
+  auto const strings_count = strings.size();
+  if (strings_count == 0) return cudf::make_empty_column(cudf::type_id::BOOL8);
+
+  CUDF_EXPECTS(prefix.is_valid(stream), "Parameter prefix must be valid.");
+
+  auto const d_prefix       = cudf::string_view(prefix.data(), prefix.size());
+  auto const strings_column = cudf::column_device_view::create(strings.parent(), stream);
+  auto const d_strings      = *strings_column;
+
+  auto results         = make_numeric_column(cudf::data_type{cudf::type_id::BOOL8},
+                                     strings_count,
+                                     cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
```

Comment:
I suspect that the file was not properly formated (using `clang-format`).

Reply:
It passed clang-format in the pre-commit hook